gwcs/tests/test_wcs.py

```diff
@@ -1443,3 +1444,29 @@ def test_bounding_box_is_returned_F():
 
     # Show the the bounding box is different between the two WCS objects
     assert gwcs_object_after.bounding_box != gwcs_object_before.bounding_box
+
+
+def test_no_bounding_box_if_read_from_file(tmp_path):
+    bad_wcs = gwcs_2d_bad_bounding_box_order()
+
+    # Check the waring is issued for the bounding box of this WCS object
+    with pytest.warns(wcs.GwcsBoundingBoxWarning):
+        bad_wcs.bounding_box
+
+    # Check that the warning is not issued again the second time
+    with warnings.catch_warnings():
+        warnings.simplefilter("error")
+        bad_wcs.bounding_box
+
+    # Write a bad wcs bounding box to an asdf file
+    asdf_file = tmp_path / "bad_wcs.asdf"
+    af = asdf.AsdfFile({"wcs": gwcs_2d_bad_bounding_box_order()}) # re-create the bad wcs object
+    af.write_to(asdf_file)
+
+    with asdf.open(asdf_file) as af:
+        wcs_from_file = af["wcs"]
+
+    # Check that no warning is issued for the bounding box of this WCS object
+    with warnings.catch_warnings():
+        warnings.simplefilter("error")
```

Comment:
This filter is probably unnecessary if you already turn all warnings in pytest into exception but probably also does not hurt.

Reply:
OK nvm... looks like you are not doing that.

https://github.com/spacetelescope/gwcs/blob/7f92615c221a91151379add48523f3feec5b4e6d/pyproject.toml#L87-L90